src/iceberg/manifest_list.h

```diff
@@ -0,0 +1,235 @@
+  /// Length of the manifest file in bytes
+  int64_t manifest_length;
+  /// Field id: 502
+  /// ID of a partition spec used to write the manifest; must be listed in table metadata
+  /// partition-specs
+  int32_t partition_spec_id;
+  /// Field id: 517
+  /// The type of files tracked by the manifest, either data or delete files; 0 for all v1
+  /// manifests
+  ManifestContent content;
+  /// Field id: 515
+  /// The sequence number when the manifest was added to the table; use 0 when reading v1
+  /// manifest lists
+  int64_t sequence_number;
+  /// Field id: 516
+  /// The minimum data sequence number of all live data or delete files in the manifest;
+  /// use 0 when reading v1 manifest lists
+  int64_t min_sequence_number;
+  /// Field id: 503
+  /// ID of the snapshot where the manifest file was added
+  int64_t added_snapshot_id;
+  /// Field id: 504
+  /// Number of entries in the manifest that have status ADDED (1), when null this is
+  /// assumed to be non-zero
+  std::optional<int32_t> added_files_count;
+  /// Field id: 505
+  /// Number of entries in the manifest that have status EXISTING (0), when null this is
+  /// assumed to be non-zero
+  std::optional<int32_t> existing_files_count;
+  /// Field id: 506
+  /// Number of entries in the manifest that have status DELETED (2), when null this is
+  /// assumed to be non-zero
+  std::optional<int32_t> deleted_files_count;
+  /// Field id: 512
+  /// Number of rows in all of files in the manifest that have status ADDED, when null
+  /// this is assumed to be non-zero
+  std::optional<int64_t> added_rows_count;
+  /// Field id: 513
+  /// Number of rows in all of files in the manifest that have status EXISTING, when null
+  /// this is assumed to be non-zero
+  std::optional<int64_t> existing_rows_count;
+  /// Field id: 514
+  /// Number of rows in all of files in the manifest that have status DELETED, when null
+  /// this is assumed to be non-zero
+  std::optional<int64_t> deleted_rows_count;
+  /// Field id: 507
+  /// Element field id: 508
+  /// A list of field summaries for each partition field in the spec. Each field in the
+  /// list corresponds to a field in the manifest file's partition spec.
+  std::vector<PartitionFieldSummary> partitions;
+  /// Field id: 519
+  /// Implementation-specific key metadata for encryption
+  std::vector<uint8_t> key_metadata;
+  /// Field id: 520
+  /// The starting _row_id to assign to rows added by ADDED data files
+  int64_t first_row_id;
+
+  /// \brief Checks if this manifest file contains entries with ADDED status.
+  bool has_added_files() const { return added_files_count.value_or(-1) > 0; }
+
+  /// \brief Checks if this manifest file contains entries with EXISTING status.
+  bool has_existing_files() const { return existing_files_count.value_or(-1) > 0; }
+
+  /// \brief Checks if this manifest file contains entries with DELETED status
+  bool has_deleted_files() const { return deleted_files_count.value_or(-1) > 0; }
```

Comment:
I think this is incorrect. When we don't know, we assume that there are deleted files.

```suggestion
  bool has_deleted_files() const { return deleted_files_count.value_or(-1) > 0; }
```
https://github.com/apache/iceberg/blob/4ca3f10832907540eee798236fa401eb07c7e7df/api/src/main/java/org/apache/iceberg/ManifestFile.java#L180